File: Data Structures & Algorithms/is-anagram/submission-0.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        sDict = {}
        # tDict = {}
        for c in s:
            if c in sDict:
                sDict[c] += 1
            else:
                sDict[c] = 1
        for c in t:
            if c in sDict:
                if sDict[c] > 1:
                    sDict[c] -= 1
                else:
                    sDict.pop(c)
            else:
                return False
        if len(sDict) == 0:
            return True
        return False

File: Data Structures & Algorithms/number-of-students-unable-to-eat-lunch/submission-0.cpp
class Solution {
public:
    int countStudents(vector<int>& students, vector<int>& sandwiches) {
        int sandwitchStart = 0;

        queue<int> studentsQ;

        for (int student: students) {
            studentsQ.push(student);
        }

        int orginalSize = students.size();
        while (!studentsQ.empty()) {
            for (int i = 0; i < orginalSize; i++) {
                int student = studentsQ.front();
                studentsQ.pop();
                if (sandwiches[sandwitchStart] == student) {
                    sandwitchStart++;
                }
                else {
                    studentsQ.push(student);
                }
            }
            if (orginalSize != studentsQ.size()) {
                orginalSize = studentsQ.size();
            }
            else {
                break;
            }
        }

        return studentsQ.size();
    }
};